File: APIdoc/scala/apidoc/src/main/scala/io/point/tutorials/AddStorageConnection.scala
package io.point.tutorials

import io.point.Pio
import dispatch._, Defaults._
import play.api.libs.json.{JsString, JsObject, JsArray}

/**
  * User: jconley
  * Date: 5/7/2014
  */
object AddStorageConnection {

  //populate with your site argument values
  val siteArgValues: Seq[String] = Seq()

  def main(args: Array[String]) {

    val (email, password, apiKey) = ("", "", "")

    for {
      keyJson <- Pio.authenticate(email, password, apiKey)
      sessionKey = keyJson.getOrElse(throw new Exception("Session key not found"))
      storageTypesJson <- Pio.listStorageTypes(sessionKey)
      storageTypes = storageTypesJson.getOrElse(throw new Exception("Access rules not found"))
      paramsJson <- Pio.listStorageTypeParams(sessionKey, storageTypes(0).as[JsArray].value(0).as[Int])
      params = paramsJson.getOrElse(throw new Exception("Folders not found"))
    } yield {
      val paramsJs = JsObject(params.value.zip(siteArgValues).map{ pair =>
        pair._1.as[JsArray].value(4).as[String] -> JsString(pair._2)
      })
      Pio.createStorageSite(sessionKey, storageTypes(0).as[JsArray].value(0).as[Int], "New Storage Connection", siteArguments = paramsJs)
    }
  }
}

File: APIdoc/java/apidoc/src/main/java/io/point/tutorials/AddStorageConnection.java
package io.point.tutorials;

import io.point.Pio;
import org.codehaus.jackson.JsonNode;
import org.codehaus.jackson.node.ArrayNode;
import org.codehaus.jackson.node.JsonNodeFactory;
import org.codehaus.jackson.node.ObjectNode;

import java.util.Iterator;

/**
 * User: jconley
 * Date: 5/6/2014
 */
public class AddStorageConnection {

    public static void main(String[] args) throws Exception{
        String email = "";
        String password = "";
        String apiKey = "";

        String sessionKey = authenticate(email, password, apiKey);

        ArrayNode storageTypesArray = Pio.listStorageTypes(sessionKey);
        JsonNode firstType = storageTypesArray.get(0);

        ArrayNode paramsArray = Pio.getStorageSiteParams(sessionKey, firstType.get(0).asInt());

        ObjectNode params = JsonNodeFactory.instance.objectNode();
        Iterator<JsonNode> rootObjects = paramsArray.getElements();
        while(rootObjects.hasNext()) {
            ArrayNode file = (ArrayNode)rootObjects.next();
            params.put(file.get(4).asText(), "");
        }

        Pio.addStorageSite(sessionKey, firstType.get(0).asInt(), "Joe Test", Pio.getDefaultFlags(), params);
    }
}
